fastexcel-writer/src/main/java/org/dhatim/fastexcel/Workbook.java

```diff
@@ -204,21 +205,53 @@ private void writeWorkbookFile() throws IOException {
              * (if there are any repeating rows or cols in the sheet at all) **/
     
             for (Worksheet ws : worksheets) {
+                int worksheetIndex = getIndex(ws) - 1;                
                 String defineName = Stream.of(ws.getRepeatingCols(),ws.getRepeatingRows())
                                 .filter(Objects::nonNull)
-                                .map(r -> ws.getName() + "!" + r.toString())
+                                .map(r -> "&apos;" + ws.getName() + "&apos;!" + r.toString())
                                 .collect(Collectors.joining(","));
+                
+                w.append("<definedNames>");
```

Comment:
If defineName and namesRanges are empty then <definedNames> will never be closed.

Reply:
I think you overlooked (it's being closed on line 254 either way.

Reply:
So if both `defineName` and `namedRanges` are not empty, then it will be closed twice?

Reply:
It will be opened and then closed only once, but there are conditional <defineName> tags inside of this <definedNames> tag.

Reply:
If they are empty - we'll have \<definedNames>\</definedNames>, otherwise if any of them are non-empty we will have
```
\<definedNames>
    \<definedName>....sth....\</definedName>
    \<definedName>....sth2...\</definedName>
\</definedNames>
```

Reply:
Oh. I see it now. I misstook line 221 as `</definedNames>`. Nevertheless I would be good to cover this part of code in a test.

Reply:
Okay, will try to do that. Where should I put this test?

Reply:
You can put it in a new class in `src/test/java`

Reply:
Can I put it into PoiCompatibilityTest.java ?

Reply:
Done!

> Please include a test that verifies (for example using Apache POI) that a worksheet with a named range is valid.

Done!